### DISCUSSION TITLE
v1.93.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#6481] [Announcements]
# v1.93.0

## Announcement 📢

We are running a logo design contest for Immich; please visit the [discussion thread](https://github.com/immich-app/immich/discussions/6318) and express your ideas if you have any. The contest will close on **March 4th**

## Highlights

Welcome to release `v1.93.0` of Immich. There are many updates in this version that we hope you'll like. Some of the key highlights include:

- Improved face grouping for facial detection - switch to DBSCAN clustering algorithm.
- New usage quota mechanism for users - specify the maximum storage each user can use on the server.
- Improve the deletion actions on the mobile app - Present explicit options to clear up confusion about each action.
- Faster map rendering on the mobile app.
- Force deletion with `Shift + Del` on the web app.
- Notable fix: The web can now show assets with a date in the future.

### New face clustering algorithm

To improve the accuracy of grouping faces, Immich is now using a more advanced clustering algorithm for facial recognition called DBSCAN. This algorithm introduces a concept of density, making facial recognition more precise. To better understand DBSCAN's work, please watch [this video](https://www.youtube.com/watch?v=RDZUdRSDOok) for a step-by-step visualization.

In addition to better facial recognition, a feature of this change is the ability to re-run facial recognition without re-processing each asset. This is because the Recognize Faces job has been split into two jobs: Face Detection and Facial Recognition, with only Face Detection processing images with machine learning models. Re-running Facial Recognition (say, after changing facial recognition settings) is much quicker as a result.

![image](https://user-images.githubusercontent.com/27055614/297937458-04a942c8-49f3-4041-bef2-0127af003dc8.png)

### Usage quota

You can now specify the storage quota for the user as the instance’s admin; once the limit is reached, the user won’t be able to upload to the instance anymore.

![user table](https://user-images.githubusercontent.com/27055614/297935804-eb5d3310-f552-4139-8bff-56b43e95a059.png)

<p align="center" >
<img src="https://user-images.githubusercontent.com/27055614/297935828-f2cb14e7-3934-48f6-bae3-a08578a96229.png" alt="edit quota" height="450"/>

<img src="https://github.com/immich-app/immich/assets/4334196/ab5a25d8-400d-4769-8806-74c234bde9dc" alt="edit quota" />
</p>

### Deletion actions on the mobile app

To increase the wife-approval-factor aspect of Immich, we always try to improve the UI/UX, and make the application simpler for non-tech-savvy users. To make the delete action easier to understand, we now have three separate buttons to handle different scenarios that can happen, which are presented in the table below.

![image](https://user-images.githubusercontent.com/27055614/297937107-813122c4-d1fe-4e31-ab50-e6342099716c.png)

</br>

![image](https://user-images.githubusercontent.com/27055614/297936540-5edaee35-98ef-4387-a531-c11b17aab17d.png)

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.
- - - -

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### 🗄️ Server
* refactor(server): domain and infra modules by @jrasm91 in https://github.com/immich-app/immich/pull/6301
* refactor(server): split api and jobs into separate e2e suites by @jrasm91 in https://github.com/immich-app/immich/pull/6307
* fix(server): exif gps decoding by @waclaw66 in https://github.com/immich-app/immich/pull/6138
* refactor: open api by @jrasm91 in https://github.com/immich-app/immich/pull/6334
* feat(server): track metadata extracted at by @jrasm91 in https://github.com/immich-app/immich/pull/6352
* feat(server, web): quotas by @cfitzw in https://github.com/immich-app/immich/pull/4471
* fix(server): extract image description by @tlvince in https://github.com/immich-app/immich/pull/6344
* fix(server): add filename search by @sushain97 in https://github.com/immich-app/immich/pull/6394
* fix: enable/disable password login on truenas by @jrasm91 in https://github.com/immich-app/immich/pull/6433
* fix(server): handle 5 digit years by @jrasm91 in https://github.com/immich-app/immich/pull/6457
* fix(server): set log level of immich-admin process in boostrap function by @alextran1502 in https://github.com/immich-app/immich/pull/6458
* feat(server, web): smart merge by @martabal in https://github.com/immich-app/immich/pull/5796
* feat(server): provide the ability to search archived photos by @stevenwcarter in https://github.com/immich-app/immich/pull/6332
* fix(server): scale transcoded videos if dimensions are odd by @mertalev in https://github.com/immich-app/immich/pull/6461
* feat(format): hif format by @haossu in https://github.com/immich-app/immich/pull/6477
### 📱 Mobile
* fix(mobile): ensure notifier is mounted before updating state by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6308
* fix(mobile): copy shared link by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6310
* feat(mobile): Adds GPS locator to map prompt for easy geolocation by @sirebellum in https://github.com/immich-app/immich/pull/6282
* chore(dep): remove unused badges dep by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6384
* feat(mobile): Focus search on doubletap nav button by @justinvdk in https://github.com/immich-app/immich/pull/6048
* refactor(mobile): maplibre by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6087
* chore(dep): update auto_route by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6390
* fix(mobile): null check on null value on top app bar by @alextran1502 in https://github.com/immich-app/immich/pull/6406
* feat(mobile): quota by @alextran1502 in https://github.com/immich-app/immich/pull/6409
* feat(mobile): separate delete buttons by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4505
* fix(mobile): add to album - list thumbnails by @waclaw66 in https://github.com/immich-app/immich/pull/6444
* fix(mobile): asset state when delete from trash by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6476
* feat(mobile): long-press delete button to permanently delete asset by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6240
### 🖥️ Web
* fix(web): grid on people page by @martabal in https://github.com/immich-app/immich/pull/5640
* fix(web): handle trailing slash in external domain by @danieldietzler in https://github.com/immich-app/immich/pull/6253
* fix(web): logout and clear user store when using back button on the change password form by @alextran1502 in https://github.com/immich-app/immich/pull/6288
* fix(web): auto-expand multi-line asset descriptions by @martabal in https://github.com/immich-app/immich/pull/6305
* fix(web): copy shared link by @jrasm91 in https://github.com/immich-app/immich/pull/6309
* refactor(web): admin settings by @danieldietzler in https://github.com/immich-app/immich/pull/6177
* fix(web): OAuth settings nits by @bo0tzz in https://github.com/immich-app/immich/pull/6330
* fix(web): Uses asset exif info for desc in asset viewer by @sellnat77 in https://github.com/immich-app/immich/pull/4249
* fix(web): wrap long word in description by @alextran1502 in https://github.com/immich-app/immich/pull/6351
* fix(web): quota usage view by @danieldietzler in https://github.com/immich-app/immich/pull/6358
* chore(web): quota enhancement by @alextran1502 in https://github.com/immich-app/immich/pull/6371
* Move 'Add' button on album user invite to the same row as 'To' List  by @A-wels in https://github.com/immich-app/immich/pull/6447
* feat(web): force delete with shift key by @martabal in https://github.com/immich-app/immich/pull/6239
* fix(web): trash or delete by @martabal in https://github.com/immich-app/immich/pull/6475
### 🧠 Machine Learning
* feat(ml): ARMNN acceleration by @fyfrey in https://github.com/immich-app/immich/pull/5667
* fix(ml): remove unused import by @mertalev in https://github.com/immich-app/immich/pull/6356
* feat(server): separate face clustering job by @mertalev in https://github.com/immich-app/immich/pull/5598
### ⚡ CLI
* fix(cli): uploadCounters increase only when files are uploaded by @AngelPone in https://github.com/immich-app/immich/pull/6357
### 📓 Documentation
* docs: update Chinese README by @indam in https://github.com/immich-app/immich/pull/6245
* fix(docs): Use absolute path for external library guide by @bo0tzz in https://github.com/immich-app/immich/pull/6290
* Clarify offline asset deletion process by @gtsteffaniak in https://github.com/immich-app/immich/pull/6333
* Edit and update of FAQ+new database GUI guide by @aviv926 in https://github.com/immich-app/immich/pull/5958
* fix(docs) Fix relative paths leading to broken links by @thariq-shanavas in https://github.com/immich-app/immich/pull/6354
* fix(docs) Fix command name by @Funk66 in https://github.com/immich-app/immich/pull/6368
* docs: fix pgAdmin links by @jrasm91 in https://github.com/immich-app/immich/pull/6403
* Update remove-offline-files.md by @hirenshah in https://github.com/immich-app/immich/pull/6449
* docs: remove tsc check by @jrasm91 in https://github.com/immich-app/immich/pull/6464
* chore: update bug template by @jrasm91 in https://github.com/immich-app/immich/pull/6465
### 🔨 Build
* chore: release note groups by @jrasm91 in https://github.com/immich-app/immich/pull/6297
* ci: Run e2e tests on self-hosted runner by @bo0tzz in https://github.com/immich-app/immich/pull/6296
### 🤖 Dependencies
* fix(deps): update docs by @renovate in https://github.com/immich-app/immich/pull/6284
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/6302
* chore(renovate): enforce compatible flavor for guava by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6392
* fix(deps): update dependency geo-tz to v8 by @renovate in https://github.com/immich-app/immich/pull/6388
* chore(renovate): enforce compatible flavor for guava using versionScheme by @shenlong-tanwen in https://github.com/immich-app/immich/pull/6398
* deps: fix guava versioning by @jrasm91 in https://github.com/immich-app/immich/pull/6402
* fix(deps): update dependency com.google.guava:guava to v33 by @renovate in https://github.com/immich-app/immich/pull/5390
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/6412
* chore(deps): update web by @renovate in https://github.com/immich-app/immich/pull/6413
* fix(deps): update docs by @renovate in https://github.com/immich-app/immich/pull/6420
* deps: separate out reflect metadata by @jrasm91 in https://github.com/immich-app/immich/pull/6425
* chore(deps): update dependency @types/node to v20.11.0 by @renovate in https://github.com/immich-app/immich/pull/6424
* chore(deps): update base-image to v20240111 (major) by @renovate in https://github.com/immich-app/immich/pull/6355
* chore(deps): update mambaorg/micromamba:bookworm-slim docker digest to 377aafa by @renovate in https://github.com/immich-app/immich/pull/6434
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/6415
* deps: update docs by @jrasm91 in https://github.com/immich-app/immich/pull/6462

## New Contributors
* @indam made their first contribution in https://github.com/immich-app/immich/pull/6245
* @gtsteffaniak made their first contribution in https://github.com/immich-app/immich/pull/6333
* @aviv926 made their first contribution in https://github.com/immich-app/immich/pull/5958
* @justinvdk made their first contribution in https://github.com/immich-app/immich/pull/6048
* @tlvince made their first contribution in https://github.com/immich-app/immich/pull/6344
* @A-wels made their first contribution in https://github.com/immich-app/immich/pull/6447
* @hirenshah made their first contribution in https://github.com/immich-app/immich/pull/6449
* @stevenwcarter made their first contribution in https://github.com/immich-app/immich/pull/6332
* @AngelPone made their first contribution in https://github.com/immich-app/immich/pull/6357
* @haossu made their first contribution in https://github.com/immich-app/immich/pull/6477
* @sirebellum made their first contribution in https://github.com/immich-app/immich/pull/6282

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.92.1...v1.93.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.93.0'>v1.93.0</a>.</em>

**Comment by Snuupy** [comment]
I hope there will be some tools for users to manage their libraries to clear out large and/or unused files/very similar files or to be able to sort by them, similar to https://photos.google.com/quotamanagement where you can review and delete "Large photos & videos" or czkawka's similar images/videos feature

**Reply by jrasm91** [reply]
Not a bad follow up idea to the quota feature.

**Comment by waclaw66** [comment]
What's purpose of this?
![obrazek](https://github.com/immich-app/immich/assets/15554561/b1a4efc7-9f06-41e2-8232-410431d06f40)

**Reply by alextran1502** [reply]
It is a bug apparently. Probably a hot fix soon

**Reply by alextran1502** [reply]
Are you using a config file?

**Reply by waclaw66** [reply]
.env only for DB_xxx and such stuff.

**Reply by jrasm91** [reply]
Looks like it was just broken, not sure how that was missed. #6485 adds back the label/descriptions.

**Reply by nicomerl** [reply]
<img width="867" alt="image" src="https://github.com/immich-app/immich/assets/112687734/dc4dc6c9-b353-4f3e-82ab-cdc84c27796f">

It Looks Like theres a few things in the settings broken.

**Comment by iheartpizza1** [comment]
So many great features, thank you!
I love the new way you're handling config file changes. No breaking changes on key conflicts for a few releases now. 👍

Is the image stacks feature being removed, or is this a bug with the latest Android build?
![Screenshot_20240119_135850](https://github.com/immich-app/immich/assets/5097250/3d34bd22-9ca6-4c82-922a-ae6266a8e67c)
Instead of the Stack button, I get the Upload button although the selected photos have already been uploaded.

**Reply by alextran1502** [reply]
Look like this is a bug

**Reply by alextran1502** [reply]
Cc: @shenlong-tanwen

**Reply by shenlong-tanwen** [reply]
> Cc: @shenlong-tanwen 

Ah, found the root cause. Will raise a PR soon. Thanks for the notification

**Comment by fwsmit** [comment]
I'm not entirely convinced that adding more delete actions improve the wife approval factor. There is a whole table needed to explain the feature.
Maybe it could be simplified by realizing that the action that will probably be performed most often is the "remove everywhere" action on merged or remote assets. This is a safe action, since it moves the items to thrash. I think that should be the only action with a button, instead of having three buttons. The other actions might be in a context menu

**Reply by nicomerl** [reply]
I was thinking the same. Looking at the bigger Brother Google Photos: 
- one delete Button that moves files from Immich to „Immich-Trash“ ( Auto empty trash After 30/60/90/something days)
- A „free up Space on device“-Button in the Settings that Removes all Backed up files which are still Stored on the phone. So you are not clicking the wrong Button by accident.

**Reply by alextran1502** [reply]
We are actively listening for feedback in this one. Thank you

**Reply by Chaphasilor** [reply]
+1 for the "Free up space" functionality

**Reply by fwsmit** [reply]
Yeah for freeing up space on the phone I would have a dedicated button somewhere else. This could have options like deleting photos older than X time, videos older than X time and a checkbox for keeping photos that are in albums (though that would only really be useful once the immich app is usable offline).

**Reply by fwsmit** [reply]
Freeing up space on the server is less of a priority to me, since photos don't take up a lot of space relative to what you would have on a home server. A VPS might be more tight on storage

**Comment by aisbergde** [comment]
with the separate implementation of face detection and recognition - would it be possible to run a new recognition on all non hidden face groups, which are not named? This way classified faces would keep and there will be a chance to recognize new matches.
